directory/basicdir.go

```diff
@@ -0,0 +1,168 @@
+package directory
+
+import (
+	"context"
+
+	"github.com/ipfs/go-unixfsnode/data"
+	"github.com/ipfs/go-unixfsnode/iter"
+	dagpb "github.com/ipld/go-codec-dagpb"
+	"github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/schema"
+)
+
+var _ ipld.Node = UnixFSBasicDir(nil)
+var _ schema.TypedNode = UnixFSBasicDir(nil)
+
+type UnixFSBasicDir = *_UnixFSBasicDir
+
+type _UnixFSBasicDir struct {
+	_substrate dagpb.PBNode
+}
+
+func NewUnixFSBasicDir(ctx context.Context, substrate dagpb.PBNode, nddata data.UnixFSData, _ *ipld.LinkSystem) (ipld.Node, error) {
+	if nddata.FieldDataType().Int() != data.Data_Directory {
+		return nil, data.ErrWrongNodeType{data.Data_Directory, nddata.FieldDataType().Int()}
+	}
+	return &_UnixFSBasicDir{_substrate: substrate}, nil
+}
+
+func (n UnixFSBasicDir) Kind() ipld.Kind {
+	return n._substrate.Kind()
+}
+
+// LookupByString looks for the key in the list of links with a matching name
+func (n UnixFSBasicDir) LookupByString(key string) (ipld.Node, error) {
+	links := n._substrate.FieldLinks()
+	link := lookup(links, key)
+	if link == nil {
+		return nil, schema.ErrNoSuchField{Type: nil /*TODO*/, Field: ipld.PathSegmentOfString(key)}
```

Comment:
leaving comment to flag this todo :)

Reply:
this one's actually an upstream design problem.   (When I drafted that error type, I thought having codegen spit out values for self-description of the types it had just emitted would be a "just around the corner" feature, and so I'd put a pointer to type info here.  That turned out to be misplaced optimism.  It's likely this should be replaced by a convention of type name strings instead.)

I'll make an issue upstream to at least track this.